google_gke_tenant/gke_service_account.tf

```diff
@@ -0,0 +1,33 @@
+resource "google_service_account" "gke-account" {
+  account_id  = "gke-${var.environment}"
+  description = "GKE deployments in the ${var.environment} environment"
+  project     = var.project_id
+}
+
+resource "google_service_account_iam_binding" "workload-identity-for-gke" {
+  service_account_id = google_service_account.gke-account.name
+  role               = "roles/iam.workloadIdentityUser"
+  members = [
+    "serviceAccount:${var.cluster_project_id}.svc.id.goog[${var.namespace}/external-secrets]"
+  ]
+}
+
+# permissions for use with External Secrets Operator in GKE
+#
+resource "google_project_iam_member" "sa-role-token-creator" {
+  project = var.project_id
+  role    = "roles/iam.serviceAccountTokenCreator"
+  member  = "serviceAccount:${google_service_account.gke-account.name}"
+}
+
+resource "google_project_iam_member" "sa-role-secret-viewer" {
+  project = var.project_id
+  role    = "roles/secretmanager.viewer"
```

Comment:
Same q as above, [the docs](https://external-secrets.io/v0.5.8/provider-google-secrets-manager/#creating-workload-identity-service-accounts) say:

> ROLE_NAME: should be `roles/secretmanager.secretAccessor` - so you make the pod only be able to access secrets on Secret Manager

Reply:
Ah yes, I think I had to add this to make the tag-based secrets ingestion work (since it needs to list all the secrets), but since we're not doing that now, I can remove.